File: src/main/java/MachTeacher/MachTeacher/dto/MessageDTO.java
package MachTeacher.MachTeacher.dto;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class MessageDTO {

    private Long id;

    private Long conversationId;

    private Long senderId;
    private Long receiverId;

    private Long sessionId;

    private String content;

    private LocalDateTime sentAt;

    private boolean mine;
    private boolean read;
}

File: MachTeacher/app/src/main/java/com/example/machteacher/dto/MessageDto.kt
package com.example.machteacher.dto

data class MessageDto(
    val id: Long,
    val conversationId: Long,
    val senderId: Long,
    val receiverId: Long?,
    val sessionId: Long?,
    val content: String,
    val sentAt: String,
    val mine: Boolean,
    val read: Boolean
)
